File: app/src/main/java/top/i97/editadapter/adapter/TestEditAdapter.java
package top.i97.editadapter.adapter;

import androidx.annotation.Nullable;
import android.view.View;
import android.widget.CheckBox;

import java.util.List;

import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import top.i97.editadapter.R;
import top.i97.editadapter.entity.TestBean;
import top.i97.editadapterlib.adapter.BaseQuickEditModeAdapter;
import top.i97.editadapterlib.inter.ISelected;
import top.i97.editadapterlib.util.ListUtils;

public class TestEditAdapter extends BaseQuickEditModeAdapter<TestBean, BaseViewHolder> {

    public TestEditAdapter(@Nullable List<TestBean> data) {
        super(R.layout.item_test_edit, data);
    }

    @Override
    protected void convertView(BaseViewHolder helper, TestBean item) {
        helper.setText(R.id.tvTitle, item.getTitle());
    }

    @Override
    public String getDeleteParams() {
        List<ISelected> selectedList = getSelectedList();
        if (!ListUtils.isEmpty(selectedList)) {
            StringBuilder sb = new StringBuilder();
            for (ISelected iSelected : selectedList) {
                if (iSelected instanceof TestBean) {
                    sb.append(((TestBean) iSelected).getId()).append(",");
                }
            }
            return sb.toString();
        }
        return null;
    }

    @Override
    public CheckBox getCheckBox(BaseViewHolder helper) {
        return helper.getView(R.id.checkBox);
    }

    @Override
    public View getHideView(BaseViewHolder helper) {
        return helper.getView(R.id.checkBox);
    }
}

File: app/src/main/java/top/i97/editadapter/adapter/TestEditAdapter.kt
/*
MIT License

Copyright (c) 2020 Plain

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
 */

package top.i97.editadapter.adapter

import android.view.View
import android.widget.CheckBox
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import top.i97.editadapter.entity.MyDataBean
import top.i97.editadapter.R
import top.i97.editadapterlib.adapter.BaseQuickEditModeAdapter

/**
 * 演示适配器
 *
 * @author Plain
 */
class TestEditAdapter(data: MutableList<MyDataBean>) : BaseQuickEditModeAdapter<MyDataBean, BaseViewHolder>(R.layout.item_test_edit, data) {

    override fun convertView(helper: BaseViewHolder, item: MyDataBean) {
        helper.setText(R.id.tvTitle, item.title)
    }

    override fun getCheckBox(helper: BaseViewHolder): CheckBox? {
        return helper.getView(R.id.checkBox)
    }

    override fun getHideView(helper: BaseViewHolder): View? {
        return helper.getView(R.id.checkBox)
    }

    override fun getDeleteParams(): String? {
        return getSelectedList().let {
            val stringBuilder = StringBuilder()
            for (iSelected in it) {
                if (iSelected is MyDataBean) {
                    // 拼接 `id` 为删除所需参数
                    stringBuilder.append(iSelected.title).append(",")
                }
            }
            stringBuilder.toString()
        }
    }

}